ir/inst_conversion.go

```diff
@@ -30,6 +30,11 @@ type InstTrunc struct {
 // NewTrunc returns a new trunc instruction based on the given source value and
 // target type.
 func NewTrunc(from value.Value, to types.Type) *InstTrunc {
+	fromSize := from.Type().(*types.IntType).BitSize
```

Comment:
This would not hold up for integer vectors I think.

From the https://github.com/llir/testdata repo:

> llvm/test/CodeGen/AMDGPU/store-private.ll:  `%0 = trunc <4 x i32> %in to <4 x i8>`

Reply:
Moved to #71, which handles this case.